src/main/diagnostic.ts

```diff
@@ -0,0 +1,86 @@
+import type { PortMapping } from "@common/types";
+import { NatType, Presence } from "@common/types";
+import { createPmpClient, createUpnpClient } from "@xmcl/nat-api";
+import { gateway4async } from "default-gateway";
+import Tracer from "nodejs-traceroute";
+import { createServer, request } from "stun";
+
+export async function getNatType(): Promise<{ address: string; natType: NatType }> {
+  const stunServer = createServer({ type: "udp4" });
+  const stunResponse1 = await request("stun1.l.google.com:19302", { server: stunServer });
```

Comment:
Also can you have this as a constant?
```suggestion
  const stunResponse1 = await request(GOOGLE_STUN_SERVER, { server: stunServer });
```
where you also have:
```ts
const GOOGLE_STUN_SERVER = "stun1.l.google.com:19302";
```
ealier in the file, after the imports.